sphinxgallery/docs_resolv.py

```diff
@@ -42,12 +46,28 @@ def _get_data(url):
     else:
         with open(url, 'r') as fid:
             data = fid.read()
-        fid.close()
 
     return data
 
-mem = joblib.Memory(cachedir='_build')
-get_data = mem.cache(_get_data)
+
+def get_data(url, cached_file='_build/searchindex'):
+    """Persistent dictionary usage to retrieve the search indexes"""
+
+    try:
+        if isinstance(url, unicode):
+            url = url.encode('utf-8')
+    except NameError:
+        pass
```

Comment:
This seems a bit convoluted. I have to admit I didn't know off the top of my head that unicode wasn't a builtin in python 3.

I would do something like that instead:

```python
# shelve keys need to be str in python 2
if six.PY2 and isinstance(url, unicode):
    url = url.encode('utf-8')
```

Also this should be done outside of this PR and is not really high-priority but it'd be nice to have a cache function in sphinx-gallery (similar to jolib.Memory.cache) because you may want to cache other stuff than url content at some point in the future.